File: app-components/src/main/java/org/synsystems/onlypass/components/preferences/GlobalPreferencesModule.java
package org.synsystems.onlypass.components.preferences;

import org.synsystems.onlypass.components.AppScope;

import dagger.Module;
import dagger.Provides;
import io.reactivex.Observable;

@Module
public class GlobalPreferencesModule {
  @Provides
  @AppScope
  public GlobalPreferences provideGlobalPreferences() {
    // TODO replace with actual implementation
    return new GlobalPreferences() {
      @Override
      public Observable<Pulse> observeRemoteLoggingEnabled() {
        return Observable.empty();
      }

      @Override
      public Observable<Pulse> observeRemoteLoggingDisabled() {
        return Observable.empty();
      }
    };
  }
}

File: app-components/src/main/java/org/synsystems/onlypass/components/preferences/GlobalPreferencesModule.kt
package org.synsystems.onlypass.components.preferences

import org.synsystems.onlypass.components.AppScope
import org.synsystems.onlypass.framework.rxutils.Pulse

import dagger.Module
import dagger.Provides
import io.reactivex.Observable

@Module
class GlobalPreferencesModule {
  @Provides
  @AppScope
  fun provideGlobalPreferences(): GlobalPreferences {
    // TODO replace with actual implementation
    return object : GlobalPreferences {
      override fun observeRemoteLoggingEnabled(): Observable<Pulse> {
        return Observable.empty()
      }

      override fun observeRemoteLoggingDisabled(): Observable<Pulse> {
        return Observable.empty()
      }
    }
  }
}